jbehave-support-core/src/main/java/org/jbehavesupport/core/ssh/SshTemplate.java

```diff
@@ -1,109 +1,109 @@
-        Long timeOffset = serverEpochTime - ZonedDateTime.now().toEpochSecond();
+        Long serverEpochTime = valueOf(executeCommand(Commands.EPOCH_TIME).trim());
+        Long timeOffset = serverEpochTime - now().toEpochSecond();
 
         // if server has different time zone
         String timeZone = executeCommand(Commands.TIMEZONE).trim();
-        ZonedDateTime serverStartTime = startTime.withZoneSameInstant(ZoneId.of(timeZone)).plusSeconds(timeOffset);
-        ZonedDateTime serverEndTime = endTime.withZoneSameInstant(ZoneId.of(timeZone)).plusSeconds(timeOffset);
+        ZonedDateTime serverStartTime = startTime.withZoneSameInstant(of(timeZone)).plusSeconds(timeOffset);
+        ZonedDateTime serverEndTime = endTime.withZoneSameInstant(of(timeZone)).plusSeconds(timeOffset);
 
-        List<String> logPaths = rollingLogResolver.resolveLogNames(sshSetting.getLogPath(), this, serverStartTime, serverEndTime);
+        List<String> logPaths = rollingLogResolver.resolveLogNames(sshSetting.getLogPath(), this,
+            serverStartTime, serverEndTime);
         StringBuilder logs = new StringBuilder();
         for (String logPath : logPaths) {
-            logs.append(copySingleLogFile(serverStartTime, serverEndTime, logPath));
+            logs.append(copySingleLogFile(serverStartTime, serverEndTime, timestampPosixRegex, logPath));
         }
-
         return new SshLog(logs.toString(), sshSetting);
     }
 
-    private String copySingleLogFile(final ZonedDateTime startTime, final ZonedDateTime endTime, final String logPath) throws IOException {
-        // get first and last line number
-        DateTimeFormatter formatter = DateTimeFormatter.ofPattern(timestampFormat);
-        String firstLastLineCommand = String
-            .format(Commands.FIRST_LAST_LINES_BETWEEN_TIMESTAMPS, startTime.format(formatter), endTime.format(formatter), logPath);
-        String firstLastLines = executeCommand(firstLastLineCommand).trim();
-
-        String logContents = "";
-        if (!firstLastLines.isEmpty()) {
-            String[] lineNumbers = firstLastLines.split("\\s+");
-            Long startLine = Long.valueOf(lineNumbers[0]);
-            Long endLine = Long.valueOf(lineNumbers[1]);
-
-            // get lines between line numbers
-            String getLinesCommand = String.format(Commands.PRINT_BETWEEN_LINES, startLine, endLine, logPath);
-            logContents = executeCommand(getLinesCommand).trim();
-        }
-
-        return logContents;
+    private String copySingleLogFile(final ZonedDateTime startTime,
+                                     final ZonedDateTime endTime,
+                                     final String timestampPosixRegex,
+                                     final String logPath) throws IOException {
+        DateTimeFormatter formatter = ofPattern(timestampFormat);
+        String command = copyToString(new ClassPathResource("get-log-between-timestamps-template.awk").getInputStream(),
```

Comment:
Might be better to move it to class level so the file wont get loaded each time the log copying is called.

Reply:
I addressed it. Thanks for the suggestion.